modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/model/listener/CommerceOrderItemModelListener.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.model.listener;
+
+import com.liferay.commerce.constants.CommerceOrderConstants;
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.model.CommerceOrderItem;
+import com.liferay.commerce.order.engine.CommerceOrderEngine;
+import com.liferay.commerce.util.CommerceOrderThreadLocal;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ModelListener;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Brian I. Kim
+ */
+@Component(service = ModelListener.class)
+public class CommerceOrderItemModelListener
+	extends BaseModelListener<CommerceOrderItem> {
+
+	@Override
+	public void onAfterRemove(CommerceOrderItem commerceOrderItem) {
+		try {
+			if (CommerceOrderThreadLocal.isDeleteInProcess()) {
```

Comment:
how do you check that the order being deleted here is the same order that have this order item?

Reply:
@andrea-ale-sbarra , the answer for your question is actually the same answer as the one I sued for @riccardo-alberti 's question.

We have to check whether DELETE is in progress for CommercOrder when CommerceOrder is deleted because we don't want the modelListener for CommerceOrderItem to run when CommerceOrder is deleted.